indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/pki/X509CertificateExtension.kt

```diff
@@ -39,19 +61,47 @@ data class X509CertificateExtension @Throws(Asn1Exception::class) private constr
 
     companion object : Asn1Decodable<Asn1Sequence, X509CertificateExtension> {
 
+        private val extensionDecoders: MutableMap<ObjectIdentifier, (Asn1Sequence, Asn1Element.Tag?, Boolean) -> X509CertificateExtension> = mutableMapOf(
+            KnownOIDs.basicConstraints_2_5_29_19 to BasicConstraintsExtension::decodeFromTlv,
+            KnownOIDs.nameConstraints_2_5_29_30 to NameConstraintsExtension::decodeFromTlv,
+            KnownOIDs.policyConstraints_2_5_29_36 to PolicyConstraintsExtension::decodeFromTlv,
+            KnownOIDs.certificatePolicies_2_5_29_32 to CertificatePoliciesExtension::decodeFromTlv,
+            KnownOIDs.policyMappings to PolicyMappingsExtension::decodeFromTlv,
+            KnownOIDs.inhibitAnyPolicy to InhibitAnyPolicyExtension::decodeFromTlv,
+            KnownOIDs.keyUsage to KeyUsageExtension::decodeFromTlv
+        )
+        private val mutex = Mutex()
+
+
+        suspend fun registerExtensionDecoder(
```

Comment:
Instead of making this suspend you could use a `MutableStateFlow` with `.update { ... }` to atomically update an immutable map.